src/lib/util/util.js

```diff
@@ -98,6 +99,182 @@ class Util {
+	 * @param {number} i Just an iteration count to prevent infinite loops
+	 * @returns {string}
+	 */
+	static _getObjType(values, depth, i) {
+		if (!Array.isArray(values)) throw new TypeError("You're using this function wrong; `values` must be an array");
+		if (typeof depth !== 'number') throw new TypeError('`depth` is missing, or not a number');
+		if (typeof i !== 'number') throw new TypeError('`i` is missing, or not a number');
+		// Collections have useful methods, which work on Sets.
+		const Coll = Collection.prototype;
+
+		const objTypes = new Set(values.map(val => this.getComplexType(val).type));
+		const nonNullTypes = new Set();
+		const nullTypes = new Set();
+		for (const type of objTypes.values()) {
+			if (['null', 'undefined'].includes(type)) nullTypes.add(type);
+			else nonNullTypes.add(type);
+		}
+
+		if (nonNullTypes.size > 1) return '*';
+		if (nonNullTypes.size === 1) {
+			const type = Coll.first.call(nonNullTypes);
+			const value = values.find(val => val != null); // eslint-disable-line eqeqeq
+			const nestedType = this.getComplexType(value);
+			let nestedTypeStr = '';
+			if (i < depth) {
+				if (nestedType.basicType === 'object') {
+					if (Array.isArray(value)) nestedTypeStr = this.getArrayType(value, depth, i + 1);
+					else if (value instanceof Map) nestedTypeStr = this.getMapType(value, depth, i + 1);
+					else if (value instanceof Set) nestedTypeStr = this.getSetType(value, depth, i + 1);
+					else nestedTypeStr = this.getObjectType(value, depth, i + 1);
+				} else if (nestedType.basicType === 'function') {
+					nestedTypeStr = this.getFunctionType(value, depth, i + 1);
+				}
+			}
+			if (nullTypes.size > 0) return `?${type}${nestedTypeStr}`;
+			return `${type}${nestedTypeStr}`;
+		}
+
+		// No types besides, possibly, "null" and "undefined"
+		if (nullTypes.size > 1) return 'null|undefined';
+		if (nullTypes.size === 1) return Coll.first.call(nullTypes);
+
+		// No types at all, i.e. no elements at all
+		return '';
+	}
+
+	/**
+	 * Determines whether the passed value is an Array.
+	 * @param {*} value The value to be checked.
+	 * @returns {boolean}
+	 */
+	static isThenable(value) {
+		return value && typeof value.then === 'function';
+	}
+
+	/**
+	 * Wrap a promise in a promise that will timeout in a certain amount of time.
+	 * 
+	 * Whichever promise (the inputted one or the timeout one) resolves first will have its value be
+	 * the resolved value of the returned promise.
+	 * @param {Promise} promise The promise to wrap.
+	 * @param {number} timeout How long the new promise should wait before timing out.
+	 * @returns {Promise}
+	 */
+	static timeoutPromise(promise, timeout) {
+		return Promise.race([promise, this.sleep(timeout, new this.TimeoutError('Promise timed out'))]);
+	}
+
+	/**
+	 * Present time duration in a nice way
+	 * @param {number} time A duration in milliseconds
+	 * @returns {string}
+	 */
+	static getNiceDuration(time) {
```

Comment:
Irrelevant with the Stopwatch class added. Please modify the command to use a Stopwatch instead.